marian-nmt/marian-dev
# Add --word-scores to marian-scorer

### Description
This PR adds the `--word-scores` option to marian-scorer for printing word-level scores.

Examples:

	./marian-server -c score.yml -t file.in file.out --word-scores
    -1.234479 ||| WordScores= -0.02151 -0.04798 -0.12068 -0.66128 -0.20932 -0.01034 -0.16291 -0.00046
	
	./marian-server -c score.yml -t file.in file.nbest --word-scores --n-best
	0 ||| und das war bei schlechtem Wetter . ||| WordScores= -0.02151 -0.04798 -0.12068 -0.66128 -0.20932 -0.01034 -0.16291 -0.00046 ||| F0= -1.23448 Score= -1.23448 ||| -1.23448

List of changes:
- Added `--word-scores` to marian-scorer.
- For this, I added a new RelationalLoss class that behaves like RescorerLoss, but it does not reduce the loss over time axis; counts are still reduced, to obtain proper sentence lengths.
- If used with `--summarize`, a warning will be printed that word-level scores are not displayed.
- If used with `--normalize`, only sentence scores are normalized. This is consistent with the current behavior of `marian-decoder --word-scores`

Added dependencies: none

### How to test
See added regression tests: https://github.com/marian-nmt/marian-regression-tests/pull/52

### Checklist
